server/metrics.go

```diff
@@ -148,10 +149,57 @@ func TimedAndCounted(handler http.Handler, fullPath string, method string, p pro
 	}
 }
 
-// PrometheusTimedAndCounted wraps a http.Handler with via prometheus.InstrumentHandler
+// PrometheusTimedAndCounted wraps a http.Handler with via promhttp.InstrumentHandlerCounter and promhttp.InstrumentHandlerDuration
 func PrometheusTimedAndCounted(handler http.Handler, name string) *Timer {
 	return &Timer{
-		isProm:  true,
-		handler: prometheus.InstrumentHandler(name, handler),
+		isProm: true,
+		handler: promhttp.InstrumentHandlerCounter(prometheusCounter(name),
+			promhttp.InstrumentHandlerDuration(prometheusHistogram(name), handler)),
```

Comment:
Does this means the counter wraps the latency layer or vice versa?

Reply:
counter wraps latency